tests/redirects/map_301.py

```diff
@@ -0,0 +1,143 @@
+from utils.urls import flatten, url_test
+
+import requests
+
+URLS = flatten((
+    url_test("/media/redesign/css/foo-min.css", "/static/build/styles/foo.css"),
+    url_test("/media/css/foo-min.css", "/static/build/styles/foo.css"),
+
+    url_test("/media/redesign/js/foo-min.js", "/static/build/js/foo.js"),
+    url_test("/media/js/foo-min.js", "/static/build/js/foo.js"),
+
+    url_test("/media/redesign/img.foo", "/static/img.foo"),
+    url_test("/media/img.foo", "/static/img.foo"),
+
+    url_test("/media/redesign/css.foo", "/static/styles.foo"),
+    url_test("/media/css.foo", "/static/styles.foo"),
+
+    url_test("/media/redesign/js.foo", "/static/js.foo"),
+    url_test("/media/js.foo", "/static/js.foo"),
+
+    url_test("/media/redesign/fonts.foo", "/static/fonts.foo"),
+    url_test("/media/fonts.foo", "/static/fonts.foo"),
+
+    url_test("/media/uploads/demos/foobar123",
+             "/docs/Web/Demos_of_open_web_technologies/",
+             status_code=requests.codes.found),
```

Comment:
This fails against ``--base_url=localhost:8000``.  It expects a 302 Found, but gets a 301 Moved Permanently.